### PR TITLE
Samplerate options

### PR DESCRIPTION
**Pull request by eljeff**
Adds a samplerate (r/R) option to the command line. Used the bit-depth (b/B) option as reference. Tried to maintain conventions and keep code clean. Please let me know if I need to change anything.

Thanks

### CODE REVIEW

File: sample-shrinker.sh [hunk 1]
```diff
@@ -90,6 +90,10 @@ Originally created to reduce the stress of streaming multiple simultaneous
     Valid values are '8' and '16'
     Default: ${minimum_bit_depth:-(none)}
 
+-s SAMPLERATE
```
Comment (colonel-blimp) [inline sample-shrinker.sh]: The args code doesn't accept `-s`―but it now accepts `-r` and `-R` , which are undocumented

File: sample-shrinker.sh [hunk 2]
```diff
@@ -251,6 +255,32 @@ prep_bitdepth_convert()
...
+{
+  # Prepare samplerate conversion
+  if [[ $samplerate -ne $target_samplerate ]]; then
+    change_summary+="${samplerate}      "
+    if [[ $samplerate -gt $target_samplerate ]]; then
+      change_summary+="${samplerate}->${target_samplerate}"
+      dst_args+=(--rate="$target_samplerate")
+
+    # Raise below-minimum samplerate samples to minimum samplerate (default: 11025)
+    elif (( samplerate < target_samplerate && samplerate < 11025 )); then
```
Comment (colonel-blimp) [inline sample-shrinker.sh]:
```suggestion
    elif (( samplerate < target_samplerate && samplerate < ${minimum_samplerate:-11025} )); then
```

File: sample-shrinker.sh [hunk 3]
```diff
@@ -251,6 +255,32 @@ prep_bitdepth_convert()
...
+      if [[ -n $minimum_samplerate ]]; then
+        dst_args+=(--rate="$minimum_samplerate")
+
+        change_summary+="${samplerate}->$minimum_samplerate+M"
```
Comment (colonel-blimp) [inline sample-shrinker.sh:282]: This is the same symbol as a minimum bit depth change gets... how should we distinguish them?  Maybe a two-character code, like `+Mr` and `+Md`?  :thinking:

File: sample-shrinker.sh [hunk 4]
```diff
@@ -456,6 +491,20 @@ while getopts 'b:B:c:x:paA:Sd:lo:nvh' opt; do
       fi
       minimum_bit_depth="${OPTARG}"
       ;;
+    r)
+      if ! [[ "$OPTARG" =~ ^[0-9]+$ ]]; then
+        .error "-r must be an integer; got invalid value: '$OPTARG'"
+        exit 1
+      fi
```
Comment (colonel-blimp) [inline sample-shrinker.sh:505]: Should we also support `44.1k` like sox does?  (We don't have to)

File: sample-shrinker.sh [hunk 5]
```diff
@@ -35,7 +35,7 @@ Caveats:
 - Samples are CONVERTED IN PLACE
   - The original file is backed up under '$backup_dir/' (change with -d)
   - Backups include spectrogram .pngs for old & new files (disable with -S)
-- Only samples that DON'T meet the target critera will be changed
+- Only samples that DONT meet the target critera will be changed
```
Comment (colonel-blimp) [inline sample-shrinker.sh]:
```suggestion
- Only samples that DO NOT meet the target critera will be changed
```

File: sample-shrinker.sh [hunk 6]
```diff
@@ -134,7 +138,7 @@ Originally created to reduce the stress of streaming multiple simultaneous
 
 -n
    Dry run
-   Log any actions that would be taken, but don't actually do anything
+   Log any actions that would be taken, but dont actually do anything
```
Comment (colonel-blimp) [inline sample-shrinker.sh]:
```suggestion
   Log any actions that would be taken, but do not actually do anything
```

**Review by colonel-blimp — COMMENTED** [review]
Thanks for the patch!

I've noted some things that might be problems.  If you don't have time to look at them, don't worry―at some point (likely the next time I use the script), I'll probably merge and deal with them

File: sample-shrinker.sh (hunk 1 above)
Comment (eljeff) [reply sample-shrinker.sh]: Remnant from earlier before I switched to r - great catch thank you!

File: sample-shrinker.sh (hunk 2 above)
Comment (eljeff) [reply sample-shrinker.sh]: This minimum stuff is what I was most unsure about - feel free to continue to push in the right direction or rip it all out if I got it wrong

File: sample-shrinker.sh (hunk 3 above)
Comment (eljeff) [reply sample-shrinker.sh:282]: Wasn't sure what the M was for :] - it just means 'modified' or what?

File: sample-shrinker.sh (hunk 4 above)
Comment (eljeff) [reply sample-shrinker.sh:505]: Could - but out of scope for what I wanted to do this round. I do like the simplicity of integers

File: sample-shrinker.sh (hunk 5 above)
Comment (eljeff) [reply sample-shrinker.sh]: easy peasy done thanky

File: sample-shrinker.sh (hunk 2 above)
Comment (colonel-blimp) [reply sample-shrinker.sh]: Actually... I notice bit depth has the same problem (probably why this does, too). :smile:    They can both wait until a later update

File: sample-shrinker.sh (hunk 3 above)
Comment (colonel-blimp) [reply sample-shrinker.sh:282]: In my head, `+` meant "modified" and `M` meant "**M**inumum," but its really just an arbitrary code to denote what will/is being changed.  At the time, the only minimum to consider was bit-depth.

Comment (eljeff) [reply sample-shrinker.sh:282]: How do you trigger / see this anyway?
Happy to leave it or set it to anything you'd like - just let me know.

**Pull request merged**